vivarium/core/emitter.py

```diff
@@ -262,6 +265,8 @@ def emit(self, data: Dict[str, Any]) -> None:
                 if key not in ['time']}
             data_at_time = assoc_path({}, self.embed_path, data_at_time)
             self.saved_data.setdefault(time, {})
+            data_at_time = _dict_to_bson(data_at_time, False, self.codec_options)
+            data_at_time = _bson_to_dict(data_at_time, self.codec_options)
```

Comment:
Can we use the `serialize.py` API here? As written, doing so would unnecessarily call `get_codec_options()` each time, so maybe let `serialize()` take in a list of codecs as an optional argument? If the codecs aren't provided, it could run `get_codec_options()`. Alternatively just run `get_codec_options()` a lot if it's not too expensive

Reply:
Good call. Addressed in 740a7ab.